ci/build/build-docker-images.py

```diff
@@ -356,11 +356,23 @@ def build_or_pull_base_images(
 
 def prep_ray_ml():
     root_dir = _get_root_dir()
-    requirement_files = glob.glob(
-        f"{_get_root_dir()}/python/**/requirements*.txt", recursive=True
-    )
-    for fl in requirement_files:
-        shutil.copy(fl, os.path.join(root_dir, "docker/ray-ml/"))
+
+    requirements_files = [
+        "python/requirements.txt",
+        "python/requirements_test.txt",
+        "python/requirements/ml/requirements_ml_docker.txt",
+        "python/requirements/ml/requirements_dl.txt",
+        "python/requirements/ml/requirements_tune.txt",
+        "python/requirements/ml/requirements_rllib.txt",
+        "python/requirements/ml/requirements_train.txt",
+        "python/requirements/ml/requirements_upstream.txt",
+    ]
+
+    for requirement_file in requirements_files:
+        shutil.copy(
+            os.path.join(root_dir, requirement_file),
+            os.path.join(root_dir, "docker/ray-ml/"),
+        )
```

Comment:
Can we add a quick sanity check here.

Maybe something like:

```python
requirements_files = ["python/requirements.txt", "python/requirements_test.txt"]
ml_requirements_files = [...] # The ml requirements files

regex = re.compile(requirements*.txt)
all_ml_requirement_files = [file for file os.walk(os.path.join(root_dir, "python/ml"))[2] if regex.match(file)]
if not len(ml_requirements_files) == len(all_requirement_files):
    raise RuntimeError("New requirements file added to ML requirements without also including it in the ML image")
```

Reply:
Looks good, I'll add this in the next revision